### PR TITLE
fix: use the right hash and add extension install test

### PR DESCRIPTION
**Pull request by gforsyth**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Seemingly arbitrary short hash length, but that's on me for not checking.  Also adding a test to make sure we can resolve extensions

### CODE REVIEW

**Comment by dhirschfeld** [comment]
Looks like it's just `linux_aarch64` where the test is failing:
```
===== testing package: python-duckdb-0.5.0-py310h335d3f6_2 =====
running run_test.py
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/python-duckdb_1663098898602/test_tmp/run_test.py", line 15, in <module>
    main()
  File "/home/conda/feedstock_root/build_artifacts/python-duckdb_1663098898602/test_tmp/run_test.py", line 11, in main
    con.install_extension("postgres_scanner")
duckdb.IOException: IO Error: Failed to download extension "postgres_scanner" at URL "[http://extensions.duckdb.org/109f932c4/linux_arm64/postgres_scanner.duckdb_extension.gz"](http://extensions.duckdb.org/109f932c4/linux_arm64/postgres_scanner.duckdb_extension.gz%22)
Extension "postgres_scanner" is an existing extension.
Are you using a development build? In this case, extensions might not (yet) be uploaded.
Tests failed for python-duckdb-0.5.0-py310h335d3f6_2.tar.bz2 - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

**Comment by gforsyth** [comment]
I'm checking in with the duckdb devs to see if they build extensions for that architecture. Might have to add a skip

**Comment by gforsyth** [comment]
DuckDB is not currently building linux-aarch64 extensions but it's on their roadmap.  Skipping the extension test for those for the moment.

xref https://github.com/duckdb/duckdb/issues/4710

File: recipe/meta.yaml [hunk 1]
```diff
@@ -63,10 +63,14 @@ test:
...
   source_files:
     - tools/pythonpkg/tests
+  files:
+    - test_extension_install.py
   commands:
     - pip check
     # ppc64le/aarch64 is emulated and fails to run the test suite here
     - pytest tools/pythonpkg/tests  # [not (ppc64le or aarch64)]
+    # DuckDB not currently building extensions for linux-aarch64
+    - python test_extension_install.py  # [not linux_aarch64]
```
Comment (mariusvniekerk) [inline recipe/meta.yaml:73]:
```suggestion
    - python test_extension_install.py  # [not (linux and aarch64)]
```

**Comment by gforsyth** [comment]
Ok, I think this worked (currently failing the non-existent linux-aarch64 extension test) -- going to merge.

**Pull request merged**